aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,297 @@
+
+.. code-block:: c#
+
+    .UseUrls("http://*:5000;http://localhost:5001;https://hostname:5002")
+
+Startup Assembly ``string``
+    Key: ``startupAssembly``. Determines the assembly to search for the ``Startup`` class. Set using the ``UseStartup`` method. Note this setting is overridden by ``WebHostBuilder.UseStartup<StartupType>``.
+
+.. code-block:: c#
+
+    .UseStartup("StartupAssemblyName")
+
+
+.. _web-root-setting:
+
+Web Root ``string``
+    Key: ``webroot``. If not specified, Web Root will use the Content Root path. Determines default root location for static files to be served by the app (relative to Content Root). Set using ``UseWebRoot``.
+
+.. code-block:: c#
+
+    .UseWebRoot("public")
+
+Use :doc:`configuration` to set configuration values to be used by the host. These values may be subsequently overridden. This is specified using ``UseConfiguration``.
+
+.. code-block:: c#
+  :emphasize-lines: 3-6,9
+
+  public static void Main(string[] args)
+  {
+    var config = new ConfigurationBuilder()
+      .AddCommandLine(args)
+      .AddJsonFile("hosting.json", optional: true)
+      .Build();
+
+    var host = new WebHostBuilder()
+      .UseConfiguration(config)
+      .UseKestrel()
+      .Configure(app => 
+      {
+        app.Run(async (context) => await context.Response.WriteAsync("Hi!"));
+      })
+    .Build();
+
+    host.Run();
+  }
+
+In the example above, command line arguments may be passed in to configure the host, or configuration settings may optionally be specified in a *hosting.json* file. To specify the host run on a particular URL, you could pass in the desired value from the command line:
+
+.. code-block:: none
+
+  dotnet run --urls "http://*:5000"
+
+The ``Run`` method starts the web app and blocks the calling thread until the host is shutdown.
+
+.. code-block:: c#
+
+    host.Run();
+
+You can run the host in a non-blocking manner by calling its ``Start`` method:
+
+.. code-block:: c#
+
+  using (host)
+  {
+    host.Start();
+    Console.ReadLine();
+  }
+
+Pass a list of URLs to the ``Start`` method and it will listen on the URLs specified:
+
+.. code-block:: c#
+
+  var urls = new List<string>() { 
+    "http://*:5000", 
+    "http://localhost:5001"
+    };
+  var host = new WebHostBuilder()
+    .UseKestrel()
+    .UseStartup<Startup>();
+  
+  using (host)
+  {
+    host.Start(urls.ToArray());
+    Console.ReadLine();
+  }
+
+You can configure services with :doc:`dependency-injection` and these services will be available in the ``Startup`` class constructor and ``Configure`` method. The following configures :doc:`logging <logging>` and a custom ``IFormatter`` service, and requests and uses these services from ``Startup``.
```

Comment:
I've removed that section. I blogged about it recently as well - we could link to that if nothing else if we think it's worth mentioning currently. http://ardalis.com/logging-and-using-services-in-startup-in-aspnet-core-apps